pkg/network/http/model_linux.go

```diff
@@ -0,0 +1,198 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package http
+
+import (
+	"encoding/hex"
+	"strings"
+	"unsafe"
+)
+
+/*
+#include "../ebpf/c/http-types.h"
+*/
+import "C"
+
+const (
+	HTTPBatchSize  = int(C.HTTP_BATCH_SIZE)
+	HTTPBatchPages = int(C.HTTP_BATCH_PAGES)
+	HTTPBufferSize = int(C.HTTP_BUFFER_SIZE)
+)
+
+type ebpfHttpTx C.http_transaction_t
+type httpNotification C.http_batch_notification_t
+type httpBatch C.http_batch_t
+type httpBatchKey C.http_batch_key_t
+
+func toHTTPNotification(data []byte) httpNotification {
+	return *(*httpNotification)(unsafe.Pointer(&data[0]))
+}
+
+// Prepare the httpBatchKey for a map lookup
+func (k *httpBatchKey) Prepare(n httpNotification) {
+	k.cpu = n.cpu
+	k.page_num = C.uint(int(n.batch_idx) % HTTPBatchPages)
+}
+
+// Path returns the URL from the request fragment captured in eBPF with
+// GET variables excluded.
+// Example:
+// For a request fragment "GET /foo?var=bar HTTP/1.1", this method will return "/foo"
+func (tx *ebpfHttpTx) Path(buffer []byte) ([]byte, bool) {
```

Comment:
Rather than have this function be defined in multiple places, I think we should remove this function from the http transaction interface, replace it with `ReqFragment() []byte` and have the http statkeeper handle extracting the path from the fragment (using the `getPath` function)

Reply:
@derekwbrown is this one of the changes which your follow-up PR is going to address?

Reply:
it should ;)